AUTO-204/AUTO-205/AUTO-206 - Automated tests for db cleanup validations - stable/5.0

nasharatm opened the pull request:
Covers: encrypted, unencrypted volumes/workloads as well. 
workload: default and scheduled 
=========
tempest.api.workloadmgr.workload.test_workload.WorkloadTest.test_3_create_workload 1
tempest.api.workloadmgr.workload.test_workload.WorkloadTest.test_12_create_scheduled_workload 1

snapshots: full, incr
=========
tempest.api.workloadmgr.snapshot.test_bootfromvol_fullsnapshot 1
tempest.api.workloadmgr.snapshot.test_snapshots 2

restore: selective, in-place, oneclick
=========
tempest.api.workloadmgr.restore.test_tvault1040_oneclick_restore 1
tempest.api.workloadmgr.barbican.test_barbican.WorkloadsTest.test_17_barbican 3

[db-cleanup-wl-snapshots-restore.pdf](https://github.com/trilioData/tempest/files/9392318/db-cleanup-wl-snapshots-restore.pdf)
[barbican-dbcleanup-all-restores-results.pdf](https://github.com/trilioData/tempest/files/9392319/barbican-dbcleanup-all-restores-results.pdf)

nasharatm committed:
Changes for adding oneclick restores

nasharatm committed:
Merge branch 'trilioData:stable/4.2' into stable/4.2

nasharatm committed:
CHanges for shared security group restore

nasharatm committed:
Added test link

nasharatm committed:
Added test case automation for multi-level shared security group

nasharatm committed:
Changes as per review comments

nasharatm committed:
Changes made for test and fix

nasharatm committed:
Merge branch 'stable/4.2' of https://github.com/trilioData/tempest into stable/4.2

nasharatm committed:
compute version change

nasharatm committed:
Changes for os-2031 and sg fix

nasharatm committed:
Change for volume type rbd

nasharatm committed:
Merge branch 'trilioData:stable/4.2' into stable/4.2

nasharatm committed:
Made changes as per review comments

nasharatm committed:
Merge branch 'trilioData:stable/4.2' into stable/4.2

nasharatm committed:
Merge branch 'trilioData:stable/4.2' into stable/4.2

nasharatm committed:
Fixes for barbican failed tests

nasharatm committed:
automated test for os-2031, volume booted

nasharatm committed:
Merge branch 'stable/5.0' of https://github.com/trilioData/tempest into stable/5.0

nasharatm committed:
Merge branch 'trilioData:stable/5.0' into stable/5.0

nasharatm committed:
Merge branch 'trilioData:stable/5.0' into stable/5.0

nasharatm committed:
Merge branch 'trilioData:stable/5.0' into stable/5.0

nasharatm committed:
fix for luks vol for rbd device cleanup

nasharatm committed:
Merge branch 'trilioData:stable/5.0' into stable/5.0

nasharatm committed:
Merge branch 'trilioData:stable/5.0' into stable/5.0

nasharatm committed:
Changes for openstack-client version

nasharatm committed:
Minor change

nasharatm committed:
Merge branch 'trilioData:stable/5.0' into stable/5.0

nasharatm committed:
Changes for db cleanup validations tests

nasharatm committed:
Removed commented code

savitha-p commented on tempest/api/workloadmgr/base.py:
```diff
@@ -4240,3 +4240,58 @@ def remove_role_from_user_project(self, project_id, user_id, role_id):
+        workload_validations = {}
+        LOG.debug("Getting list for db cleanup workload validations.")
+
+        workload_tables = ["workload_vms", "workload_vm_metadata", "scheduled_jobs", "snapshots"]
```
Move this list to tvaultconf.py

nasharatm replied:
done

savitha-p commented on tempest/api/workloadmgr/base.py:
```diff
@@ -4240,3 +4240,58 @@ def remove_role_from_user_project(self, project_id, user_id, role_id):
+        snapshot_validations = {}
+        LOG.debug("Getting list for db cleanup snapshot validations.")
+
+        snapshot_tables = ["snapshots", "snapshot_metadata", "vm_recent_snapshot", "snapshot_vm_resources",
```
Move this tables list to tvaultconf.py

nasharatm replied:
done

savitha-p commented on tempest/api/workloadmgr/base.py:
```diff
@@ -4240,3 +4240,58 @@ def remove_role_from_user_project(self, project_id, user_id, role_id):
+        restore_validations = {}
+        LOG.debug("Getting list for db cleanup restore validations.")
+
+        restore_tables = ["restore_metadata", "restored_vm_metadata", "restored_vm_resource_metadata", "restored_vm_resources", "restored_vms", "restores"]
```
Move this tables list to tvaultconf.py

nasharatm replied:
done

savitha-p commented on tempest/api/workloadmgr/workload/test_workload.py:
```diff
@@ -132,10 +135,21 @@ def test_3_create_workload(self):
                 reporting.set_test_script_status(tvaultconf.FAIL)
-
+            
+            self.db_cleanup_workload_validations()
```
Is this call required ?

nasharatm replied:
removed.

savitha-p commented on tempest/api/workloadmgr/workload/test_workload.py:
```diff
@@ -652,12 +669,24 @@ def test_12_create_scheduled_workload(self):
                 LOG.error("Workload schedule not enabled")
+        
+            # Delete workload
+            self.workload_delete(self.wid)
```
When you are explicitly deleting the workload, you need to pass workload_cleanup=False in workload_create() call.

nasharatm replied:
Added change

nasharatm committed:
Changes for wl_dbs for specific validations

nasharatm committed:
Changes for db tables per ids

nasharatm commented:
Added changes as per review comment and modified queries.
PFA the run results with latest code changes.

[1-dbcleanup-barbican-results.pdf](https://github.com/trilioData/tempest/files/9413663/1-dbcleanup-barbican-results.pdf)
[2-dbcleanup-snapshot-wl-restore-results.pdf](https://github.com/trilioData/tempest/files/9413664/2-dbcleanup-snapshot-wl-restore-results.pdf)

nasharatm committed:
Merge branch 'trilioData:stable/5.0' into stable/5.0

nasharatm committed:
Removed commented code, added few debug statements

savitha-p commented on tempest/api/workloadmgr/snapshot/test_snapshots.py:
```diff
@@ -49,6 +49,7 @@ def test_1_create_full_snapshot(self):
             global volume_id
             global workload_id
             global snapshot_id
+            global full_snapshot_validations_before
```
where is this global variable getting used ?

nasharatm replied:
It was used previously, removed as no longer needed.

nasharatm committed:
Removed unwanted variable
